allennlp/data/iterators/language_modeling_iterator.py

```diff
@@ -0,0 +1,79 @@
+from typing import Iterable, Iterator
+import logging
+
+from allennlp.data.instance import Instance
+from allennlp.data.iterators.basic_iterator import BasicIterator
+from allennlp.data.iterators.data_iterator import DataIterator, TensorDict
+from allennlp.data.dataset import Batch
+
+logger = logging.getLogger(__name__)  # pylint: disable=invalid-name
+
+
+@DataIterator.register("language_modeling")
+class LanguageModelingIterator(BasicIterator):
```

Comment:
I think it makes sense to give this object a more general name - it is useful for any problem where batching is done in the `DatasetReader` (e.g. one's with the issue I raise in #2828). Maybe something like `StraightThroughIterator` or `TrivialIterator`?